@uportal/esco-content-menu/src/components/EscoContentMenu.vue

```diff
@@ -0,0 +1,204 @@
+<template>
+  <div class="esco-content-menu" :style="'background-color:' + backgroundColor">
+    <div class="wrapper":style="'background-image: linear-gradient(0deg, rgba(0,0,0,.2),rgba(0,0,0,.2)), url(' + backgroundImg + ');'">
+      <div></div>
+      <content-favorites :portlets="portlets" :favorites="favorites" :call-after-action="actionToggleFav" :is-small="isSmall"></content-favorites>
+    </div>
+    <content-grid :portlets="portlets" :favorites="favorites" :call-after-action="actionToggleFav" :is-small="isSmall"></content-grid>
+  </div>
+</template>
+
+<script>
+  import ContentFavorites from "./ContentFavorites";
+  import ContentGrid from './ContentGrid.vue'
+  import oidc from "@uportal/open-id-connect";
+
+  const checkStatus = function(response) {
+//console.log("check response ", response);
+    if (response.ok) {
+      return response;
+    } else {
+      let error = new Error(response.statusText);
+      error.response = response;
+      throw error;
+    }
+  };
+
+  const parseJSON = function(response) {
+//console.log("Parse response for json ", response);
+    return response.json();
+  };
+
+  export default {
+    name: 'EscoContentMenu',
+    props: {
+      backgroundColor: { type: String, default: '#f3f3f3'}
+    },
+    components: {
+      ContentFavorites,
+      ContentGrid
+    },
+    data() {
+      return {
+        favorites: [],
+        portlets: [],
+        backgroundImg: 'https://lycees.netocentre.fr//annuaire_images/logos/77534903800072/logoportail1.jpg?1',
+        isSmall: false
+      }
+    },
+    mounted() {
+      this.$nextTick(function() {
+        window.addEventListener("resize", this.isXs);
+        this.isXs();
+        this.fetchPortlets();
+        this.fetchFavorites();
+      });
+    },
+    created() {
+
+    },
+    methods: {
+      getWindowWidth: function() {
+        console.log("window size ", this.$el.clientWidth);
+        return this.$el.clientWidth;
+      },
+      isXs: function() {
+        this.isSmall = this.getWindowWidth() < 620;
+      },
+      fetchPortlets() {
+        if (process.env.NODE_ENV === "development") {
+          let data = require("@gip-recia/content-grid/src/assets/browseable.json");
+          this.portlets.push(...data.portlets);
+        } else {
+          oidc()
+            .then(token => {
+              const options = {
+                method: "GET",
+                credentials: "same-origin",
+                headers: {
+                  Authorization: "Bearer " + token.encoded,
+                  "Content-Type": "application/json"
+                }
+              };
+              fetch(
+                process.env.VUE_APP_PORTAL_BASE_URL +
```

Comment:
Is this something that could be valuable to customize via web component properties/attributes?

Reply:
I have to define properties as a default prop value initialized with these env properties